compiler/rustc_mir/src/interpret/operator.rs

```diff
@@ -77,9 +77,9 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         let res = match bin_op {
             Eq => l == r,
             Ne => l != r,
-            Lt => l < r,
+            Lt => !l & r, // equals l < r
```

Comment:
Shouldn't there be an allow here instead of the new code?

1. regardless whether you think whether `l < r` is clearer than `!l & r` or not, this specific piece of code concerns the implementation of `l < r` in the constant evaluator. I think it's most clear to implement `l < r` via `l < r` instead via something that clippy thinks is better.

2. If `l < r` should be replaced with not and and operations, why not `l <= r` as well?

The [bool comparison](https://rust-lang.github.io/rust-clippy/master/#bool_comparison) lint description in clippy's docs makes sense, `x < true` is code smell. But `x < y` in an implementation of `x < y` is IMO not.

Reply:
Alright, I removed changes from the pr.
Sprinkling the rustc code with clippy allow attributes is probably something most people would object though.